src/coffee/stockimport.coffee

```diff
@@ -371,8 +371,18 @@ class StockImport
             .fetch()
             .then (result) =>
               @_updateInventory(entry, result.body, tryCount + 1)
+            .catch (err) =>
+              if (err.statusCode == 404)
+                debug "Stock not existing anymore (probably was deleted), create a new stock"
+                @client.inventoryEntries.create(entry)
+              else
+                debug "Error on refetching stock on 409 error. Details: #{JSON.stringify(err)}"
+                Promise.reject err
           else
-            Promise.reject new Error("Retry limit #{max409Retries} reached for stock #{JSON.stringify(entry)}")
+            Promise.reject new Error("Retry limit #{@max409Retries} reached for stock #{JSON.stringify(entry)}")
+        else if (err.statusCode == 404)
+          debug "Inventory was deleted during trying to update the inventory"
```

Comment:
For the sake of consistency use Inventory or stock but not both. Suggestion:
`It seem that our stock update has conflicted with parallel stock deletion process so we can not update it anymore and should rather issue create request.` It is not short but at least explains why we have this logic here. Same goes for another 404 message above.

Reply:
https://github.com/sphereio/sphere-stock-import/pull/69/commits/f0227ce01505b32c5565cc6d784389281b01533d